File: tournaments/subtractionByRegrouping/subtractionByRegrouping.py
def subtractionByRegrouping(minuend, subtrahend):
    result = []
    memory = 0;
    
    while 0 < subtrahend:
        m = minuend % 10 - memory
        s = subtrahend % 10
        
        if m >= s:
            result.append(m)
            memory = 0
        else:
            result.append(10 + m)
            memory = 1
        
        minuend //= 10
        subtrahend //= 10
    
    return result

File: tournaments/subtractionByRegrouping/subtractionByRegrouping.js
function subtractionByRegrouping(minuend, subtrahend) {
    const n = String(minuend).length;
    const result = new Array(n);

    for (let i = 0; n > i; ++i) {
        result[i] = minuend % 10;
        minuend = ~~(minuend / 10);
    }

    for (let i = 0; n > i; ++i) {
        if (subtrahend%10 > result[i]) {
            result[i] += 10;
            --result[i + 1];
        }
        subtrahend = ~~(subtrahend / 10);
    }

    return result;
}
